middleware/third_party/arduino/hardware/arduino/mt7697/libraries/MCS/src/MCS.cpp

```diff
@@ -787,7 +787,7 @@ void MCSControllerHex::_dispatch(const String& params)
 bool MCSControllerHex::_update(const String& params)
 {
     String b = params;
-    String v = b;
+    long v = strtol(&b[0], NULL, 16);
```

Comment:
you can use String::c_str() to get char* pointer
WString.h
const char* c_str() const { return buffer; }